src/HotChocolate/Core/src/Types/Resolvers/IPureResolverContext.cs

```diff
@@ -62,6 +62,14 @@ public interface IPureResolverContext : IHasContextData
     /// </returns>
     T Parent<T>();
 
+    /// <summary>
+    /// Gets the previous (parent) resolver result in its raw form (without conversions).
+    /// </summary>
+    /// <returns>
+    /// Returns the previous (parent) resolver result.
+    /// </returns>
+    object? RawParent { get; }
```

Comment:
It will work, if it's actually an object array, but `Parent` also tries converting if the parent is not an object array. If it's not an object array, it should just fail fast instead.

Reply:
We expect the parent to be an object array and nothing else at that point. The parent with the cast should be used when we want either a cast or a conversion.

Reply:
If you know it's an object array, do you need to fail fast? 
Unless we have usecases for this elsewhere, i dont think its a good idea. Changing the resolver context interface for a one of usecase should be avoided. 

@michaelstaib  do we have other places where this would be usefull? Is it worth it?

Reply:
Well the change is additive, it won't break other code (unless they provide a custom implementation of that interface). And having access to the unconverted parent is imo more useful than to its converted form. Is `Parent` only used when resolving those method calls that you showed?